io-native/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -129,4 +156,19 @@ public static Object close(BObject channel) {
         }
         return null;
     }
+
+    public static Object closeBufferedReader(BObject channel) {
+        try {
+            if(channel.getNativeData(BUFFERED_READER_ENTRY) != null) {
+                BufferedReader bufferedReader = (BufferedReader)
+                        channel.getNativeData(BUFFERED_READER_ENTRY);
+                bufferedReader.close();
+            }
+        } catch (ClosedChannelException e) {
+            return IOUtils.createError("channel already closed.");
```

Comment:
Lets have a boolean at the bal Iterable class, that way we can check for already closed channels quite quickly. Check https://github.com/ballerina-platform/module-ballerina-sql/blob/master/sql-ballerina/src/sql/types.bal#L350

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/67def2fcc0c5f6f8119178883ea13414f7c7c2e1

Reply:
How is the isClosed property updated? I dont see any return values from closeRecordReader() native function

Reply:
My bad. Fixed in here https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/6c430e760d3696fd347e6548ce22f201ed49bd84